source/slang/slang.cpp

```diff
@@ -75,39 +77,106 @@ Session::Session()
     addBuiltinSource(hlslLanguageScope, "hlsl", getHLSLLibraryCode());
 }
 
+static String getString(ISlangBlob* blob)
+{
+    if (blob)
+    {
+        size_t size = blob->getBufferSize();
+        if (size > 0)
+        {
+            const char* contents = (const char*)blob->getBufferPointer();
+            // Check it has terminating 0, if not we must construct as if it does
+            if (contents[size - 1] == 0)
+            {
+                size --;
+            }
+            return String(contents, contents + size);
+        }
+    }
+    return String();
+}
+
 struct IncludeHandlerImpl : IncludeHandler
```

Comment:
Not something this change should be responsible for fixing, but obviously the current `IncludeHandler` and `IncludeHandlerImpl` are just code bloat that serves no purpose at this point. They were introduced when I thought our API should follow the lead of fxc, but at this point we are trying to hew closer to dxc's narrower interface.

Reply:
As touched on in earlier comment it would be nice to not have IncludeHandler and IncludeHandlerImpl, but doing so pushes yet more semantics/functions out into the interface. Probably a new method to search, and a way to handle search path's would be necessary.